docs/node_setup.md

```diff
@@ -1,127 +1,128 @@
-```bash
-sudo ntpq -p
-```
+* OS: Ubuntu 20.04
+* CPU: Intel Core i7-7700K
+* Memory: 64GB RAM
+* Storage: NVMe SSD with a capacity of at least 200GB (the data footprint will grow over time)
 
-#### Firewal requirements 
+:::note
 
-* `30333` *allowed* - p2p port have to be exposed publicly to allow other nodes to connect to your node
-* `9944`  *optional* - rpc websocket port, expose this port only if want to allow external apps connect to your node (e.g https://polkadot.js.org/apps/) 
-* `9933`  *optional* - http rpc port, expose this port only if want to allow external http req to your node
+These are the minimum technical requirements which have been verified by the team. Want to make sure that your machine has sufficient resources to run a node? Run a [performance benchmark](/performance_benchmark) to find out.
 
-:::caution
-We highly recommend **TO NOT** expose RPC ports (`9944, 9933`) publicly if you are running node as validator. This ports can be abused by 3-rd party to manipulate with your node and cause slashing. 
 :::
 
 
+### 01 Check whether your system clock is synchronized
 
-### Getting binaries 
+Before running a node, you should make sure that your system clock is synchronized - this is important because validators work together. On Ubuntu 20.04 the system clock should be synchronized by default. To verify, run the following command and check the output:
 
-#### Download binaries
+```bash
+$ timedatectl | grep "System clock"
+System clock synchronized: yes
+```
 
-You can download released binaries from our [github](https://github.com/galacticcouncil/HydraDX-node/releases).
+If the output is different, then you can install NTP manually and verify again that your system clock is synchronized:
 
-#### Building binary from source
+```bash
+$ apt install ntp
+$ ntpq -p
+```
+
+### 02 Adjust your firewall settings
+Port `30333` is used for peer-to-peer connections with other nodes. If you are running the node as a validator, this is the only port we recommend exposing in your firewall.
+
+If you are *not* running the node as a validator, you can also consider exposing `9944` (for RPC websocket connections with external apps) and `9933` (for HTTP requests to your node).
+
+### 03 Download or build a binary
+You can download a binary of our latest release on [github](https://github.com/galacticcouncil/HydraDX-node/releases).
+
+Alternatively, you can build the binary from source:
 
-:::important
-Please use `git tags` with lates version *(e.g v2.0.0)* if you are building binary from souce.
-:::
 ```bash
-git clone git@github.com:galacticcouncil/HydraDX-node.git
-cd HydraDX-node/
-git checkout v2.0.0 #this is latest release in the time of writing this doc. Please check github for latest release tag before running this cmd
+# Fetch source
+$ git clone https://github.com/galacticcouncil/HydraDX-node.git
+$ cd HydraDX-node/
+
+# Make sure you are on latest release
+$ git checkout v2.0.0
```

Comment:
we have stable branch now. We can `git checkout stable` or even `git clone https://github.com/galacticcouncil/HydraDX-node.git -b stable`

Reply:
nice, much better than specifying version

Reply:
What do you think about "latest" tag @lumir-mrkva ?